api/scanner/media_encoding/executable_worker/ffmpeg_cli.go

```diff
@@ -0,0 +1,95 @@
+package executable_worker
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+
+	"github.com/photoview/photoview/api/utils"
+	"gopkg.in/vansante/go-ffprobe.v2"
+)
+
+type FfmpegCli struct {
+	path       string
+	videoCodec string
+}
+
+func newFfmpegCli() *FfmpegCli {
+	if utils.EnvDisableVideoEncoding.GetBool() {
+		log.Printf("Executable worker disabled (%s=%q): ffmpeg\n", utils.EnvDisableVideoEncoding.GetName(), utils.EnvDisableVideoEncoding.GetValue())
+		return nil
+	}
+
+	path, err := exec.LookPath("ffmpeg")
+	if err != nil {
+		log.Println("Executable worker not found: ffmpeg")
+		return nil
+	}
+
+	version, err := exec.Command(path, "-version").Output()
+	if err != nil {
+		log.Printf("Error getting version of ffmpeg: %s\n", err)
+		return nil
+	}
+
+	codec := utils.EnvVideoCodec.GetValue()
```

Comment:
No. I only test `h264` and `h264_qsv`. It should work with all `h264` codec, but not others.

We could add another ENV for encoding parameters for other codecs. It will increase the complex for users.

Another way, we change the ENV to `PHOTOVIEW_HARDWARE_ACCELERATION` and limit the value with ``/`qsv` which we are tested with.

Reply:
> I only test `h264` and `h264_qsv`. It should work with all `h264` codec, but not others.

It would be great if you can manually test also Mac HW acceleration, as Jellyfin's Ffmpeg should support it.
For other possible accelerations, we might need to wait for feedback from users, as we don't have that HW to test...

> We could add another ENV for encoding parameters for other codecs

That is how I implemented my version, but I see the complexity level and potential place for user mistakes here, so let's wait for test results on different HW and see how we can handle that in the simplest way for users...

> ENV to `PHOTOVIEW_HARDWARE_ACCELERATION` and limit the value with the empty string or `qsv`

This would mean that we limit HW support to only QSV protocol, which I'd like to avoid, as we integrated Jellyfin's Ffmpeg to be able to configure many HW acceleration protocols

Reply:
> We could add other values like `vaapi` after we verify it on the env.

As we don't have other envs, I'd like to release it with no limitations to let users test them. We need to provide a clear end-2-end guide on how to configure Photoview on the Docker host to use HW acceleration with different protocols. Of course, we can give links to vendors' pages and give some assumptions for HW, we can't test.

Reply:
`jellyfin-ffmpeg`/`arm64` doesn't work on macOS. All dependencies can't be installed.

Since a docker image can't be used on macOS, if users want to run on macOS (like Mac mini), they should compile the code and prepare dependencies separately. Otherwise, they could run the official docker image in a VM, which of course can't use hardware acceleration.

I suggest not considering macOS with the docker image. If necessary, we could provide a document about macOS.

Reply:
Ok, then it makes sense to document this case.
Once we're ready to provide the HW acceleration feature to users (merge all related PRs and close the #652  issue as completed), we definitely need to update the Readme and document a short description and a list of requirements for its activation on supported environments

Reply:
Cool) I like this implementation more)

Reply:
This PR works well in my local env. Resolved. Let me know if you have other concerns.